lib/internal/freelist.js

```diff
@@ -1,24 +1,25 @@
 'use strict';
 
-// This is a free list to avoid creating so many of the same object.
-exports.FreeList = function(name, max, constructor) {
-  this.name = name;
-  this.constructor = constructor;
-  this.max = max;
-  this.list = [];
-};
-
-
-exports.FreeList.prototype.alloc = function() {
-  return this.list.length ? this.list.pop() :
-                            this.constructor.apply(this, arguments);
-};
+class FreeList {
+  constructor(name, max, constructor) {
+    this.name = name;
+    this.constructor = constructor;
```

Comment:
Naming own properties of instances `constructor` is kind of confusing, maybe you could change that while you’re here?